core/src/subgraph/instance_manager.rs

```diff
@@ -230,6 +230,9 @@ impl<S: SubgraphStore> SubgraphInstanceManager<S> {
             (manifest, manifest_idx_and_name)
         };
 
+        let static_filters = self.static_filters
+            || (manifest.network_name() == "bsc" && manifest.templates.len() > 10000);
```

Comment:
Isn't `manifest.templates.len()` just the number of templates in the manifest? We want to count actual dynamic data sources, so want to look at `data_sources.len()` ?

Reply:
Yeah I was trying to avoid including the data_sources but you're right, it's prolly a few in 10000 so that should be fine, I'll change this

Reply:
actually this also uses the previously stored data sources, so I need to trim those as well